File: Daily Challenge/Python/Middle Print Till Repeat 7.12.2025.py
s = input().strip()
n = len(s)
mid = n // 2

result = s[mid]
visited = {s[mid]}

offset = 1
while True:
    if mid - offset < 0 or mid + offset >= n:
        break
    
    left_char = s[mid - offset]
    result += left_char
    if left_char in visited:
        break
    visited.add(left_char)
    
    right_char = s[mid + offset]
    result += right_char
    if right_char in visited:
        break
    visited.add(right_char)
    
    offset += 1

print(result)

File: Daily Challenge/C++/Middle Print Till Repeat 7.12.2025.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    cin >> s;
    int n = s.length();
    int mid = n / 2;
    
    string result = "";
    result += s[mid];
    unordered_set<char> visited = {s[mid]};
    
    int offset = 1;
    while (true) {
        if (mid - offset < 0 || mid + offset >= n) break;
        
        char left_char = s[mid - offset];
        result += left_char;
        if (visited.count(left_char)) break;
        visited.insert(left_char);
        
        char right_char = s[mid + offset];
        result += right_char;
        if (visited.count(right_char)) break;
        visited.insert(right_char);
        
        offset++;
    }
    
    cout << result << endl;
    return 0;
}